java/statistical-query-group-sort/src/main/java/com/esri/arcgisruntime/sample/statisticalquery/MainActivity.java

```diff
@@ -0,0 +1,436 @@
+    mGroupRecyclerView = findViewById(R.id.groupFieldRecyclerView);
+    mOrderByRecyclerView = findViewById(R.id.orderFieldRecyclerView);
+  }
+
+  /**
+   * Create recycler views and their adapters.
+   */
+  private void createRecyclerViews() {
+    // field type recycler view
+    mStatisticDefinitionsAsStringsList = new ArrayList<>();
+    mStatisticsDefinitionRecyclerView.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
+    mStatisticsDefinitionAdapter = new RecyclerViewAdapter(this, mStatisticDefinitionsAsStringsList, false);
+    mStatisticsDefinitionRecyclerView.setAdapter(mStatisticsDefinitionAdapter);
+
+    // group field recycler view which just takes the full list of of field names
+    mGroupRecyclerView.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
+    mGroupAdapter = new RecyclerViewAdapter(this, mFieldNameList, true);
+    mGroupRecyclerView.setAdapter(mGroupAdapter);
+
+    // order by field recycler view
+    mOrderByList = new ArrayList<>();
+    mOrderByRecyclerView.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
+    mOrderByAdapter = new RecyclerViewAdapter(this, mOrderByList, false);
+    mOrderByRecyclerView.setAdapter(mOrderByAdapter);
+  }
+
+  /**
+   * Helper method to get a list of the fields which are checked in the group fields recycler view.
+   *
+   * @return list of checked fields
+   */
+  private ArrayList<String> getCheckedFields() {
+
+    ArrayList<String> checkedFields = new ArrayList<>();
+
+    // add all checked fields in the group fields list to the query parameters
+    for (int i = 0; i < mGroupAdapter.getCheckedList().length; i++) {
+      if (mGroupAdapter.getCheckedList()[i]) {
+        checkedFields.add(mFieldNameList.get(i));
+      }
+    }
+    return checkedFields;
+  }
+
+  /**
+   * Helper method to get the sort order from a string containing a field and sort order.
+   *
+   * @param fieldAndOrder string from recycler view
+   * @return SortOrder (either ASCENDING or DESCENDING)
+   */
+  private static QueryParameters.SortOrder getSortOrderFrom(String fieldAndOrder) {
+    String orderString = fieldAndOrder.substring(fieldAndOrder.indexOf('(') + 1, fieldAndOrder.indexOf(')'));
+    QueryParameters.SortOrder sortOrder;
+    switch (orderString) {
+      case "DESCENDING":
+        sortOrder = QueryParameters.SortOrder.DESCENDING;
+        break;
+      case "ASCENDING":
+        sortOrder = QueryParameters.SortOrder.ASCENDING;
+        break;
+      default:
+        Log.e(TAG, "Invalid sort order: " + orderString);
+        sortOrder = null;
+        break;
+    }
+    return sortOrder;
+  }
+
+  /**
+   * Helper method to get a field from a string.
+   *
+   * @param fieldAndOrder string from recycler view
+   * @return field as a string
+   */
+  private static String getFieldFrom(String fieldAndOrder) {
+    return fieldAndOrder.substring(0, fieldAndOrder.indexOf(' '));
+  }
+
+  @Override protected void onPause() {
```

Comment:
I think you also need onResume and onDestroy

Reply:
onPause is only there to dismiss the alert. We use onResume and onDestroy to dispose of the MapView, but there isn't a mapview in this sample.